### PR TITLE
MEN-6717 - fix: aligned env checks with diverged setup instructions for on-prem installs

### PR DESCRIPTION
**Pull request by mzedel**
Ticket: MEN-6717
Changelog: None

### CODE REVIEW

File: src/js/selectors/index.js [hunk 1]
```diff
@@ -340,10 +340,9 @@ export const getTenantCapabilities = createSelector(
   ) => {
     const canDelta = isEnterprise || plan === PLANS.professional.id;
     const hasAuditlogs = isAuditlogEnabled && (!isHosted || isEnterprise || plan === PLANS.professional.id);
-    const hasDeviceConfig = hasAddons || (isDeviceConfigEnabled && (!isHosted || addons.some(addon => addon.name === 'configure' && Boolean(addon.enabled))));
-    const hasDeviceConnect =
-      hasAddons || (isDeviceConnectEnabled && (!isHosted || addons.some(addon => addon.name === 'troubleshoot' && Boolean(addon.enabled))));
-    const hasMonitor = hasAddons || (isMonitorEnabled && (!isHosted || addons.some(addon => addon.name === 'monitor' && Boolean(addon.enabled))));
+    const hasDeviceConfig = hasAddons || isDeviceConfigEnabled || addons.some(addon => addon.name === 'configure' && Boolean(addon.enabled));
+    const hasDeviceConnect = hasAddons || isDeviceConnectEnabled || addons.some(addon => addon.name === 'troubleshoot' && Boolean(addon.enabled));
+    const hasMonitor = hasAddons || isMonitorEnabled || addons.some(addon => addon.name === 'monitor' && Boolean(addon.enabled));
```
Comment (tranchitella) [inline src/js/selectors/index.js:345]: I could be wrong but this change will make "hasDeviceConfig" true on HM independently from the addons status.
Am I right? See: https://github.com/mendersoftware/saas/blob/master/kubernetes/mender_stack/gui/mender-gui.yaml#L87

**Review by tranchitella — CHANGES_REQUESTED** [review]

File: src/js/selectors/index.js (hunk 1 above)
Comment (mzedel) [reply src/js/selectors/index.js:345]: then we can remove these from there I guess...

Comment (mzedel) [reply src/js/selectors/index.js:345]: https://github.com/mendersoftware/saas/pull/1069

Comment (tranchitella) [reply src/js/selectors/index.js:345]: so... if I get it correctly, you use the env variables for OS and rely on add-ons for ENT.
Right?

Comment (mzedel) [reply src/js/selectors/index.js:345]: from now on: yes
the only exception being that `HAVE_ADDONS` flag as I'm not sure where it is used (the slack messages from Marcin mentioned the backend tests, but from searching the backend repos I can't find it)...

Comment (tranchitella) [reply src/js/selectors/index.js:345]: @mzedel `DEVICEAUTH_HAVE_ADDONS` and `USERADM_HAVE_ADDONS` maybe?

Comment (mzedel) [reply src/js/selectors/index.js:345]: these might be what became of it - in the messages he referred to `HAVE_ADDONS` in particular and wanted it added to the UI... but I'd also be happy to remove it and see if something would actually fail - it doesn't seem to be used anymore for customer communication

Comment (mzedel) [reply src/js/selectors/index.js]: `HAVE_ADDONS` should now also be removed.

**Comment by tranchitella** [comment]
Reject, we don't want to go down this path.

**Pull request closed without merging**